solr/core/src/java/org/apache/solr/util/PackageTool.java

```diff
@@ -360,15 +366,19 @@ private String getZkHost(CommandLine cli) throws Exception {
     String zkHost = cli.getOptionValue("zkHost");
     if (zkHost != null) return zkHost;
 
-    String systemInfoUrl = solrUrl + "/admin/info/system";
-    CloseableHttpClient httpClient = SolrCLI.getHttpClient();
+    Http2SolrClient http2SolrClient = getHttpSolrClient(solrUrl);
```

Comment:
I think this could be:

```
try (var http2SolrClient = getHttpSolrClient(solrUrl)) {
...
}
```

then you don't even need the finally?

Reply:
agreed...   @risdenk I wonder if there is a way to add a rule that would ensure we all follow the try/with/resources model?